karbon/karbon_cluster_registration_service_test.go

```diff
@@ -118,6 +115,81 @@ func TestKarbonCreateClusterRegistration(t *testing.T) {
 	// TODO get task uuid status
 }
 
+func TestKarbonCreateClusterRegistrationWithNoCategory(t *testing.T) {
+	interceptor := khttpclient.NewInterceptor(http.DefaultTransport)
+	creds := testhelpers.CredentialsFromEnvironment(t)
+	nkeClient, err := NewKarbonAPIClient(creds, WithRoundTripper(interceptor))
+	require.NoError(t, err)
+
+	kctx := mock.NewContext(mock.Config{
+		Mode: keploy.MODE_TEST,
+		Name: t.Name(),
+	})
+
+	test_cluster_name := strings.ToLower("cluster2")
+	test_cluster_uuid := strings.ToLower("E33FD0FD-5673-45FA-825D-7EF869A91577")
+	test_metadata_apiversion := "v1.1.0"
+	test_metadata := &Metadata{APIVersion: &test_metadata_apiversion}
+
+	t.Log("Get Cluster registration")
+	responseGetReg, err := nkeClient.ClusterRegistrationOperations.GetK8sRegistration(kctx, test_cluster_uuid)
+	if err == nil {
+		t.Log("Get Cluster registration exists")
+		validateK8sClusterRegistrationGetResponse(t, test_cluster_name, test_cluster_uuid, responseGetReg)
+		// Registration exists. delete it so that we can create it
+		t.Log("Delete Cluster registration")
+		responseDelReg, err := nkeClient.ClusterRegistrationOperations.DeleteK8sRegistration(kctx, test_cluster_uuid)
+		assert.NoError(t, err)
+		validateK8sClusterRegistrationDeleteResponse(t, test_cluster_name, test_cluster_uuid, responseDelReg)
+	}
+
+	t.Log("Create Cluster registration")
+	createRequest := &K8sCreateClusterRegistrationRequest{
+		Name:     &test_cluster_name,
+		UUID:     test_cluster_uuid,
+		Metadata: test_metadata,
+	}
+
+	// check if the error is expected
+	_, err = nkeClient.ClusterRegistrationOperations.CreateK8sRegistration(kctx, createRequest)
+	if assert.Error(t, err) {
+		assert.Equal(t, NoCategoryMappingError, err)
+	}
+}
+
+func TestKarbonCreateClusterRegistrationWithNoUUID(t *testing.T) {
+	interceptor := khttpclient.NewInterceptor(http.DefaultTransport)
+	creds := testhelpers.CredentialsFromEnvironment(t)
+	nkeClient, err := NewKarbonAPIClient(creds, WithRoundTripper(interceptor))
+	require.NoError(t, err)
+
+	kctx := mock.NewContext(mock.Config{
+		Mode: keploy.MODE_TEST,
+		Name: t.Name(),
+	})
+
+	test_cluster_name := strings.ToLower("cluster3")
+	test_category_mapping := map[string]string{
+		fmt.Sprintf("kubernetes-io-cluster-%s", test_cluster_name): "owned",
+		"KubernetesClusterName": test_cluster_name,
+	}
+	test_metadata_apiversion := "v1.1.0"
+	test_metadata := &Metadata{APIVersion: &test_metadata_apiversion}
+
+	t.Log("Create Cluster registration")
+	createRequest := &K8sCreateClusterRegistrationRequest{
+		Name:              &test_cluster_name,
+		CategoriesMapping: test_category_mapping,
+		Metadata:          test_metadata,
+	}
+
+	// check if the error is expected
+	_, err = nkeClient.ClusterRegistrationOperations.CreateK8sRegistration(kctx, createRequest)
+	if assert.Error(t, err) {
+		assert.Equal(t, NoUUIDError, err)
```

Comment:
Do a `assert.Contains` instead. Exact error matching leads to brittle assertions.